server/test/services/applicant/ApplicantServiceTest.java

```diff
@@ -882,7 +883,7 @@ public void getReadOnlyApplicantService_getsReadOnlyApplicantServiceForTheApplic
   }
 
   @Test
-  public void submitApplication_returnsSavedApplication() {
+  public void submitApplication_returnsSavedApplication() throws ProgramNotFoundException {
```

Comment:
suggestion (blocking): Please add a test for the `submitApplication` method that tests the three eligibility situations you're handling in `ApplicantService`: eligible, ineligible, and not computed. You might able to handle this by adding some asserts to existing tests.